src/tree/ContainerAppItem.ts

```diff
@@ -154,7 +154,12 @@ export async function getContainerEnvelopeWithSecrets(context: IActionContext, s
 
     // verify all top-level properties
     for (const key of Object.keys(containerAppEnvelope)) {
-        containerAppEnvelope[key] = nonNullProp(containerAppEnvelope, <keyof ContainerApp>key);
+        // We know it's undefined but we want it to throw the error
+        if (containerAppEnvelope[key] === undefined) {
```

Comment:
Are there other falsy values that we should check for? `null`, `''` (empty string), `0`, etc.? Or is `undefined` the only one we expect to deal with here?

Reply:
Currently the error is occurring because the workloadProfileType property is set to null which is why we changed it to just check for undefined. I don't think we would want to check for '' or 0 in this case either.

Reply:
Yeah, that function is only here to typecheck all of the properties because when dealing with them in other commands, it's annoying to always do =|| or assert that they're non-null.